### PR TITLE
Editor Can Leave Feedback And Select Appropriate Articles For Publication

### PR DESCRIPTION
**Pull request by rupertlion**
## PT Link

https://www.pivotaltracker.com/story/show/161311353

## PR Includes

* Select option for status ("approved", "rejected" or "commented" (ensure status attribute changing in article model) 
* Adding comment attribute to article model
* Adding comment field and updating comment attribute in model
* Disabling comment field when revision button not selected
* Adding seeds for pending articles
* Simple styling of admin review (editor) page

***NOTE: Task [Create option if published as premium or not (add attribute to article model)] moved to separate PR to reduce current PR size***

## Wireframe
image

### CODE REVIEW

File: app/views/admin/articles/index.html.haml [hunk 1]
```diff
@@ -9,3 +9,5 @@
             %p= article.body
         #imageplaceholder
             = image_tag("trump.jpg")
+        = check_box_tag 'approve', 'approve'
```
Comment (angiecstn) [inline app/views/admin/articles/index.html.haml]: I might be wrong, but is the 'approve' repetition needed?

File: Gemfile [hunk 2]
```diff
@@ -27,6 +27,7 @@ group :development, :test do
   gem 'coveralls', require: false
   gem 'chromedriver-helper'
   gem 'selenium-webdriver'
+  gem 'launchy'
```
Comment (tochman) [inline Gemfile]: This one is not really needed. With Selenium, if you want to inspect what the page looks like in the browser, you just insert a `Then stop` step and the execution will halt. At that point you can inspect the elements and run any js code in the console (as long as it does not affect the back-end)

**Review by tochman — COMMENTED** [review]
I would also suggest that you add some specs to your `article_spec` for statuses:
```ruby
  describe 'Statuses' do 
      it { is_expected.to respond_to :pending! }
      it { is_expected.to respond_to :pending? }
      it { is_expected.to respond_to :approved! }
      it { is_expected.to respond_to :approved? }
       it { is_expected.to respond_to :rejected! }
      it { is_expected.to respond_to :rejected? }
    end
```

And since `enum` gives us scopes 'for free' we should also test those class methods:

```ruby
  describe 'Class methods' do
    it { expect(described_class).to respond_to :pending }
    it { expect(described_class).to respond_to :approved }
    it { expect(described_class).to respond_to :rejected }
  end
```

This is the basics. If you want to be more detailed, you could create a few objects using `let!` and make sure that the scopes are returning an expected result. (See the admin/articles_controller#index action for an example of usage of the `Article.pending`

File: app/controllers/admin/articles_controller.rb [hunk 3]
```diff
@@ -19,9 +19,20 @@ def create
     end
   end
 
+  def update
+    @article = Article.find(article_params)
+    @article.send([params[:status], '!'].join.to_sym)
+    binding.pry
```
Comment (angiecstn) [inline app/controllers/admin/articles_controller.rb]: Can't the binding.pry be removed?

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -26,6 +26,11 @@
   click_on button
 end
 
+Then("I select {string}") do |value|
+  choose value
+end
+
 Then("stop") do
   binding.pry
```
Comment (angiecstn) [inline features/step_definitions/basic_steps.rb:34]: binding.pry can be removed

**Review by angiecstn — APPROVED** [review]

**Pull request merged**